test/e2e_node/e2e_remote.go

```diff
@@ -0,0 +1,178 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e_node
+
+import (
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"math/rand"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	"github.com/golang/glog"
+	"os/user"
+)
+
+// to run on gce --ssh-options ""
+var sshOptions = flag.String("ssh-options", "", "Commandline options passed to ssh.")
+var sshEnv = flag.String("ssh-env", "", "Use predefined ssh options for environment.  Options: gce")
+
+var sshOptionsMap map[string]string
+
+func init() {
+	usr, err := user.Current()
+	if err != nil {
+		glog.Fatal(err)
+	}
+	sshOptionsMap = map[string]string{
+		"gce": fmt.Sprintf("-i %s/.ssh/google_compute_engine -o UserKnownHostsFile=/dev/null -o IdentitiesOnly=yes -o CheckHostIP=no -o StrictHostKeyChecking=no", usr.HomeDir),
+	}
+}
+
+// CreateTestArchive builds the local source and creates a tar archive e2e_node_test.tar.gz containing
+// the binaries k8s required for node e2e tests
+func CreateTestArchive() string {
+	// Build the executables
+	BuildGo()
+
+	// Build the e2e tests into an executable
+	glog.Infof("Building ginkgo k8s test binaries...")
+	testDir, err := getK8sNodeTestDir()
+	if err != nil {
+		glog.Fatalf("Failed to locate test/e2e_node directory %v.", err)
+	}
+	out, err := exec.Command("ginkgo", "build", testDir).CombinedOutput()
+	if err != nil {
+		glog.Fatalf("Failed to build e2e tests under %s %v.  Output:\n%s", testDir, err, out)
+	}
+	ginkgoTest := filepath.Join(testDir, "e2e_node.test")
+	if _, err := os.Stat(ginkgoTest); os.IsNotExist(err) {
+		glog.Fatalf("Failed to locate test binary %s", ginkgoTest)
+	}
+	defer os.Remove(ginkgoTest)
+
+	// Make sure we can find the newly built binaries
+	buildOutputDir, err := getK8sBuildOutputDir()
+	if err != nil {
+		glog.Fatalf("Failed to locate kubernetes build output directory %v", err)
+	}
+	kubelet := filepath.Join(buildOutputDir, "kubelet")
+	if _, err := os.Stat(kubelet); os.IsNotExist(err) {
+		glog.Fatalf("Failed to locate binary %s", kubelet)
+	}
+	apiserver := filepath.Join(buildOutputDir, "kube-apiserver")
```

Comment:
I could, but then it would use the k8sBinDir if specified, which would be weird since it wouldn't be using the binary it had just built.  Thoughts?